web/pdf_page_view.js

```diff
@@ -654,31 +687,49 @@ class PDFPageView {
     redrawAnnotationEditorLayer = false,
     redrawXfaLayer = false,
     redrawTextLayer = false,
+    postponeDrawingDelay = -1,
   }) {
     // Scale target (canvas or svg), its wrapper and page container.
-    const width = this.viewport.width;
-    const height = this.viewport.height;
-    const div = this.div;
-    target.style.width =
-      target.parentNode.style.width =
-      div.style.width =
-        Math.floor(width) + "px";
-    target.style.height =
-      target.parentNode.style.height =
-      div.style.height =
-        Math.floor(height) + "px";
-    // The canvas may have been originally rotated; rotate relative to that.
-    const relativeRotation =
-      this.viewport.rotation - this.paintedViewportMap.get(target).rotation;
-    const absRotation = Math.abs(relativeRotation);
-    let scaleX = 1,
-      scaleY = 1;
-    if (absRotation === 90 || absRotation === 270) {
-      // Scale x and y because of the rotation.
-      scaleX = height / width;
-      scaleY = width / height;
+
+    if (target instanceof HTMLCanvasElement) {
```

Comment:
Since I've not tested in detail yet, does this branch also work correctly for the `zoomLayer` (since I'm assuming it'll also hit it)?

Reply:
What do you mean exactly ?
Except in the svg case, only a canvas is passed to `cssTransform` and the `zoomLayer` is its parent.

Reply:
I was just wondering if the zoomLayer is "special" in some way, since I wasn't entirely sure how it worked (it was a long time since I looked at its code).